File: ruby/send.rb

require 'rubygems'
require 'stomp'
require 'json'

conn = Stomp::Connection.open(nil, nil, 'host', '61613', true)

headers = {'eventtype' => 'ping', 'timestamp' => Time.new.to_i * 1000 } # timestamp in milliseconds
conn.publish '/queue/events', JSON({:host => 'example.com', :idle => 42}), headers

File: perl/send.pl
use strict;
use JSON;
use Net::Stomp;
use warnings;

my $stomp = Net::Stomp->new({ hostname => 'localhost', port => '61613' });
my $headers = { timestamp => time*1000, destination => '/queue/events', eventtype => "memory"};
my $body = to_json({free => 1234});

my $frame = Net::Stomp::Frame->new({
        command => 'SEND',
        headers => $headers,
        body => $body,
});

$stomp->connect;
$stomp->send_frame($frame);
$stomp->disconnect;
